clang/test/SemaSYCL/free_function_kernel_params.cpp

```diff
@@ -0,0 +1,59 @@
+// RUN: %clang_cc1 -internal-isystem %S/Inputs -fsycl-is-device -ast-dump \
+// RUN: %s -o - | FileCheck %s
+// UNSUPPORTED: system-windows
+// This test checks parameter rewriting for free functions with parameters
+// of type scalar and pointer.
+// Windows support will be added later.
```

Comment:
Why does it not work on Windows? Name mangling? Parameter generation shouldn't really depend on platform.

Reply:
It does work on Windows. I made the name check generic and enabled the test on windows.

Reply:
The free function kernel naming is currently the same on Linux and Windows (just as it is for normal SYCL kernels). So kernel name checks in the tests are not OS-dependent. I have reenabled the tests on Windows.